pkg/serviceimport/weights/controller.go

```diff
@@ -0,0 +1,125 @@
+limitations under the License.
+*/
+package weights
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/log"
+	lighthousev1a1 "github.com/submariner-io/lighthouse/pkg/apis/lighthouse.submariner.io/v1alpha1"
+	lighthouseClientset "github.com/submariner-io/lighthouse/pkg/client/clientset/versioned"
+	lhconstants "github.com/submariner-io/lighthouse/pkg/constants"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+type NewClientsetFunc func(kubeConfig *rest.Config) (lighthouseClientset.Interface, error)
+
+// NewClientset allows indirection hook for unit tests to supply fake client sets.
+var NewClientset NewClientsetFunc
+
+type Controller struct {
+	// Indirection hook for unit tests to supply fake client sets
+	NewClientset    NewClientsetFunc
+	weightsInformer cache.Controller
+	weightsStore    cache.Store
+	stopCh          chan struct{}
+	localClusterID  string
+}
+
+func NewController(localClusterID string) *Controller {
+	return &Controller{
+		NewClientset:   getNewClientsetFunc(),
+		stopCh:         make(chan struct{}),
+		localClusterID: localClusterID,
+	}
+}
+
+func getNewClientsetFunc() NewClientsetFunc {
+	if NewClientset != nil {
+		return NewClientset
+	}
+
+	return func(c *rest.Config) (lighthouseClientset.Interface, error) {
+		clientset, err := lighthouseClientset.NewForConfig(c)
+		return clientset, errors.Wrap(err, "error creating lighthouseClientset")
+	}
+}
+
+func (c *Controller) Start(kubeConfig *rest.Config) error {
+	klog.Infof("Starting Services Import Weights Controller")
+
+	clientSet, err := c.NewClientset(kubeConfig)
+	if err != nil {
+		return errors.Wrap(err, "error creating client set")
+	}
+
+	// nolint:wrapcheck // Let the caller wrap these errors.
+	c.weightsStore, c.weightsInformer = cache.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				return clientSet.LighthouseV1alpha1().ServiceImportWeightMaps(metav1.NamespaceAll).List(context.TODO(), options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return clientSet.LighthouseV1alpha1().ServiceImportWeightMaps(metav1.NamespaceAll).Watch(context.TODO(), options)
+			},
+		},
+		&lighthousev1a1.ServiceImportWeightMap{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    func(_ interface{}) {},
+			UpdateFunc: func(_, _ interface{}) {},
+			DeleteFunc: func(_ interface{}) {},
+		},
+	)
+
+	go c.weightsInformer.Run(c.stopCh)
+
+	return nil
+}
+
+func (c *Controller) Stop() {
+	close(c.stopCh)
+
+	klog.Infof("ServiceImportsWeights Controller stopped")
+}
+
+func (c *Controller) GetWeightFor(service, namesapce, inCluster string) int64 {
+	obj, exists, err := c.weightsStore.GetByKey(lhconstants.SubmarinerWeightMapKey)
```

Comment:
So I assume the CRD is intended to be a singleton named "submariner-weights".  This lookup assumes no namespace. However if we don't make the CRD namespaced then this will work fine.

Reply:
Got it, will remove the namespace scope for the CRD - thanks!